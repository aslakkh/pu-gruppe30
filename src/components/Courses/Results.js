/**
 * Created by anderssalvesen on 08.04.2017.
 */
import React, { Component } from 'react';
import './results.css'
import {BootstrapTable, TableHeaderColumn} from 'react-bootstrap-table';
import {secondsToString, formatDate } from '../../helpers/helperFunctions';
import { removeOldGoal } from '../../helpers/auth'


let dayRowDict = {};
let weekRowDict = {};
let monthRowDict = {};

let toDelete = [];

export default class Results extends Component {

    constructor(props) {
        super(props);
        this.state = {
            course: this.props.course,
            courseID: props.courseID,
            view: props.view,
        };

    }

    componentWillReceiveProps(nextProps) {
        this.setState({
            view: nextProps.view,
            course: nextProps.course,
            courseID: nextProps.courseID,
        })
    }


    componentDidMount() {
        toDelete = [];
    }

    componentWillUnmount() {
        /*
         Iterates toDelete-list, fetches the key to oldGoal and firebase and deletes the entry in OldGoals.
         */
        for (let i = 0; i < toDelete.length; i++) {
            for(let key in dayRowDict) {
                if (key === toDelete[i]) {
                    removeOldGoal(this.state.courseID, "daily", dayRowDict[key]);
                    delete dayRowDict[key];

                }
            }
            for(let key in weekRowDict) {
                if (key === toDelete[i]) {
                    removeOldGoal(this.state.courseID, "weekly", weekRowDict[key]);
                    delete dayRowDict[key];
                }
            }
            for(let key in monthRowDict) {
                if (key === toDelete[i]) {
                    removeOldGoal(this.state.courseID, "monthly", monthRowDict[key]);
                    delete dayRowDict[key];
                }
            }
        }

    }


    createTableRows() {
        let rows = [];
        let goalList;
        if (this.state.view === "Daily Goals") {
            goalList = this.state.course.oldGoals.daily;
        } else if (this.state.view === "Weekly Goals") {
            goalList = this.state.course.oldGoals.weekly;
        } else {
            goalList = this.state.course.oldGoals.monthly;
        }

        {Object.keys(goalList).map((key) => {
            if (key != 1) {
                let date = formatDate(new Date(goalList[key].timeSet));

                let idString = date[5] + "/" + date[2] + "-" + date[0] + " " + date[6] + ":" + date[7] + ":" + date[8];

                let goal = goalList[key].goal;
                let timeSpent = goalList[key].timeSpent;
                let spanString = "";
                if (this.state.view === "Daily Goals") {
                    spanString = date[5] + "/" + date[2] + "/" + date[0];
                    dayRowDict[idString] = key;
                } else if (this.state.view === "Weekly Goals") {
                    spanString = "Week " + date[3] + ", " + date[0];
                    weekRowDict[idString] = key;
                } else {
                    spanString = date[1] + ", " + date[0];
                    monthRowDict[idString] = key;
                }

                let percent = Math.floor((timeSpent / goal) * 100);
                if (percent > 100) percent = 100;
                let goalString = secondsToString(goal);
                let timeSpentString = secondsToString(timeSpent);
                if (timeSpentString === "") timeSpentString = "-";
                rows.push({
                    id: idString,
                    set: spanString,
                    goal: goalString,
                    spent: timeSpentString,
                    percent: percent,
                })

            }
        })}
        return rows.reverse();
    }

    onAfterDeleteRow(rowKeys) {
        /*
         adds rows to delete in toDelete-list
         */

        while (rowKeys.length != 0) {
            let element = rowKeys[0];
            toDelete.push(element);
            rowKeys.splice(0,1);
        }

    }

    render() {
        const selectRowProp = {
            mode: 'checkbox',
            bgColor: '#CEE3F6',
        };

        const options = {
            afterDeleteRow: this.onAfterDeleteRow,
        };

        let rows = this.createTableRows();
        return(

            <div className="table-module">

                <BootstrapTable data={ rows } options={ options } selectRow={ selectRowProp } deleteRow={ true }>
                    <TableHeaderColumn className="column" dataField='id' isKey={ true } dataSort>Goal set</TableHeaderColumn>
                    <TableHeaderColumn className="column" dataField='set'>Span</TableHeaderColumn>
                    <TableHeaderColumn className="column" dataField='goal'>Goal</TableHeaderColumn>
                    <TableHeaderColumn className="column" dataField='spent'>Time spent</TableHeaderColumn>
                    <TableHeaderColumn className="column" dataField='percent'>% completed</TableHeaderColumn>

                </BootstrapTable>

            </div>

        );
    }
}