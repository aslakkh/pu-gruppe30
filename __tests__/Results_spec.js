/**
 * Created by jan on 20/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Results from '../src/components/Courses/Results.js';
import TestUtils from "react-addons-test-utils"
const courses={
    active: true,
        goals: {dailyGoal:{active: false,
            timeSet: 0,
            timeSpent: 0,
            value: 0},
            monthlyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0},
            weeklyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0}

        },
        oldGoals:{
        daily: {
            goal: 0,
            timeSet:0,
            timeSpent:0
        },
            weekly: {
                goal: 0,
                timeSet:0,
                timeSpent:0
            },
            monthly: {
                goal: 0,
                timeSet:0,
                timeSpent:0
            }
        }
    };
const coursesNull=null;

describe("Results", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <Results courseID={'TES1000'} course={courses} view={'Daily Goals'}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Daily", () => {
        expect(component2).toMatchSnapshot();
    });
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <Results courseID={'TES1000'} course={courses} view={'Weekly Goals'}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Weekly", () => {
        expect(component2).toMatchSnapshot();
    });
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <Results courseID={'TES1000'} course={courses} view={'Monthly Goals'}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Monthly", () => {
        expect(component2).toMatchSnapshot();
    });
});