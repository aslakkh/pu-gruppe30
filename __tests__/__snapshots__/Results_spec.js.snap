// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Daily 1`] = `
<div
  className="table-module"
>
  <BootstrapTable
    bodyContainerClass={null}
    bodyStyle={undefined}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={false}
    containerClass={null}
    containerStyle={undefined}
    csvFileName="spreadsheet.csv"
    data={
      Array [
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
      ]
    }
    deleteRow={true}
    expandColumnOptions={
      Object {
        "expandColumnBeforeSelectColumn": true,
        "expandColumnComponent": undefined,
        "expandColumnVisible": false,
      }
    }
    expandComponent={undefined}
    expandableRow={undefined}
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    headerContainerClass={null}
    headerStyle={undefined}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    keyBoardNav={false}
    maxHeight={undefined}
    multiColumnSearch={false}
    multiColumnSort={1}
    options={
      Object {
        "afterDeleteRow": [Function],
      }
    }
    pagination={false}
    printable={false}
    scrollTop={undefined}
    search={false}
    searchPlaceholder={undefined}
    selectRow={
      Object {
        "bgColor": "#CEE3F6",
        "mode": "checkbox",
      }
    }
    strictSearch={undefined}
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableHeaderClass={null}
    tableStyle={undefined}
    trClassName=""
  >
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="id"
      dataFormat={undefined}
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal set
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="set"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Span
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="goal"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="spent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Time spent
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="percent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      % completed
    </TableHeaderColumn>
  </BootstrapTable>
</div>
`;

exports[`Results Monthly 1`] = `
<div
  className="table-module"
>
  <BootstrapTable
    bodyContainerClass={null}
    bodyStyle={undefined}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={false}
    containerClass={null}
    containerStyle={undefined}
    csvFileName="spreadsheet.csv"
    data={
      Array [
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
      ]
    }
    deleteRow={true}
    expandColumnOptions={
      Object {
        "expandColumnBeforeSelectColumn": true,
        "expandColumnComponent": undefined,
        "expandColumnVisible": false,
      }
    }
    expandComponent={undefined}
    expandableRow={undefined}
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    headerContainerClass={null}
    headerStyle={undefined}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    keyBoardNav={false}
    maxHeight={undefined}
    multiColumnSearch={false}
    multiColumnSort={1}
    options={
      Object {
        "afterDeleteRow": [Function],
      }
    }
    pagination={false}
    printable={false}
    scrollTop={undefined}
    search={false}
    searchPlaceholder={undefined}
    selectRow={
      Object {
        "bgColor": "#CEE3F6",
        "mode": "checkbox",
      }
    }
    strictSearch={undefined}
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableHeaderClass={null}
    tableStyle={undefined}
    trClassName=""
  >
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="id"
      dataFormat={undefined}
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal set
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="set"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Span
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="goal"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="spent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Time spent
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="percent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      % completed
    </TableHeaderColumn>
  </BootstrapTable>
</div>
`;

exports[`Results Weekly 1`] = `
<div
  className="table-module"
>
  <BootstrapTable
    bodyContainerClass={null}
    bodyStyle={undefined}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={false}
    containerClass={null}
    containerStyle={undefined}
    csvFileName="spreadsheet.csv"
    data={
      Array [
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
        Object {
          "goal": "NaN hours and NaN minutes",
          "id": "undefined/undefined-NaN undefined:undefined:undefined",
          "percent": NaN,
          "set": "undefined, NaN",
          "spent": "NaN hours and NaN minutes",
        },
      ]
    }
    deleteRow={true}
    expandColumnOptions={
      Object {
        "expandColumnBeforeSelectColumn": true,
        "expandColumnComponent": undefined,
        "expandColumnVisible": false,
      }
    }
    expandComponent={undefined}
    expandableRow={undefined}
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    headerContainerClass={null}
    headerStyle={undefined}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    keyBoardNav={false}
    maxHeight={undefined}
    multiColumnSearch={false}
    multiColumnSort={1}
    options={
      Object {
        "afterDeleteRow": [Function],
      }
    }
    pagination={false}
    printable={false}
    scrollTop={undefined}
    search={false}
    searchPlaceholder={undefined}
    selectRow={
      Object {
        "bgColor": "#CEE3F6",
        "mode": "checkbox",
      }
    }
    strictSearch={undefined}
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableHeaderClass={null}
    tableStyle={undefined}
    trClassName=""
  >
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="id"
      dataFormat={undefined}
      dataSort={true}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal set
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="set"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Span
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="goal"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Goal
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="spent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      Time spent
    </TableHeaderColumn>
    <TableHeaderColumn
      className="column"
      columnClassName=""
      columnTitle={false}
      csvFormat={undefined}
      csvHeader={undefined}
      dataAlign="left"
      dataField="percent"
      dataFormat={undefined}
      dataSort={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filter={undefined}
      filterFormatted={false}
      filterValue={undefined}
      formatExtraData={undefined}
      headerAlign={undefined}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      onSort={undefined}
      searchable={true}
      sort={undefined}
      sortFunc={undefined}
      sortFuncExtraData={undefined}
      sortIndicator={true}
      tdAttr={undefined}
      tdStyle={undefined}
      thStyle={undefined}
      width={null}
    >
      % completed
    </TableHeaderColumn>
  </BootstrapTable>
</div>
`;
